### PR TITLE
Fix Bug 1440456 - lazy render context menus

### PR DESCRIPTION
**Pull request by rlr**

### CODE REVIEW

**Comment by rlr** [comment]
poking around in about:memory... without this patch about:newtab uses almost 10MB on a fresh new profile. with this patch, it uses about 6MB. The savings should be much more significant on a profile that fills up all the top sites and highlights.

**Comment by Mardak** [comment]
Redirecting to @sarracini !

**Comment by sarracini** [comment]
@rlr can you run the mochitests in mc for this too? I have a feeling some of the tests that check for context menu hidden might be broken

**Comment by rlr** [comment]
@sarracini mochitests are passing now

```
SUITE-END | took 17s

Overall Summary
mochitest-browser: 33/33
```

File: system-addon/test/functional/mochitest/browser_topsites_section.js [hunk 1]
```diff
@@ -23,13 +28,15 @@ test_newtab({
     await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-site-icon"),
       "Topsite tippytop icon not found");
     // There are only topsites on the page, the selector with find the first topsite menu button.
-    const topsiteContextBtn = content.document.querySelector(".top-sites-list .context-menu-button");
+    let topsiteContextBtn = content.document.querySelector(".top-sites-list .context-menu-button");
     topsiteContextBtn.click();
 
-    const contextMenu = content.document.querySelector(".top-sites-list .context-menu");
-    ok(contextMenu && !contextMenu.hidden, "Should find a visible topsite context menu");
+    await new Promise(resolve => setTimeout(resolve, 0));
```
Comment (sarracini) [inline system-addon/test/functional/mochitest/browser_topsites_section.js]: Probably better to use a ```waitForCondition``` statement here, to keep it consistent with the rest of the test:
```js
await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-sites-list .context-menu"), 
   "No context menu found");
```

**Comment by as-pine-proxy** [comment]
[Treeherder: 6adc3acb1806ad6afb4274949e7db66679763939](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=6adc3acb1806ad6afb4274949e7db66679763939)

**Pull request merged**